daemon/src/main/java/org/mvndaemon/mvnd/sync/MvndSyncContextFactory.java

```diff
@@ -80,4 +90,69 @@ public void close() {
             };
         }
     }
+
+    private static class LocalSyncContextFactory implements SyncContextFactory {
+        final Map<String, Lock> locks = new ConcurrentHashMap<>();
+        @Override
+        public SyncContext newInstance(RepositorySystemSession session, boolean shared) {
+            return new LocalSyncContext();
+        }
+
+        private class LocalSyncContext implements SyncContext {
+            private final Deque<String> locked = new ArrayDeque<>();
+
+            @Override
+            public void acquire(Collection<? extends Artifact> artifacts, Collection<? extends Metadata> metadatas) {
+                Collection<String> keys = new TreeSet<>();
+                stream(artifacts).map(this::getKey).forEach(keys::add);
+                stream(metadatas).map(this::getKey).forEach(keys::add);
+                String current = null;
+                try {
+                    for (String key : keys) {
+                        current = key;
+                        getLock(key).lock();
+                        locked.add(key);
+                    }
+                } catch (Exception e) {
+                    close();
+                    throw new IllegalStateException("Could not acquire lock for '" + current + "'" );
```

Comment:
Maybe we should not swallow the exception?

```suggestion
                    throw new IllegalStateException("Could not acquire lock for '" + current + "'", e);
```